### PR TITLE
feat(test): Sentinel Integration Test 

### PR DESCRIPTION
**Pull request by ashotland**

### CODE REVIEW

File: tests/dragonfly/sentinel_test.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    assert sentinel.slaves()[0]["port"] == str(master.port)
+
+    # Verify we can now write to replica and read replicated value from master.
+    await replica_client.set("key", "value")
+    async def get(): return await master_client.get("key") # There is no async lambda
+    await await_for(
+        get,
+        lambda val: val == b"value",
+        10, "Timeout waiting for key to exist in replica."
+    )
```
Comment (dranikpg) [inline tests/dragonfly/sentinel_test.py:160]: Sometimes generic utils and async don't mix well in Python. You could've used lambdas if you made them return coroutines which are awaited internally. Or not used the util at all:
```py
async with asyncio.timeout(10):
  while (await master_client.get('key')) != b"value":
    await asyncio.sleep(1.0)
```
But it doesn't matter as its just tests

File: tests/dragonfly/sentinel_test.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        assert subprocess.run(["docker", "rm", "-f", "-v", f"{self.container_name}"]).returncode == 0, "docker rm failed."
+
+    def run_cmd(self, args, sentinel_cmd=True, capture_output=False, assert_ok=True) -> subprocess.CompletedProcess:
+        run_args = ["redis-cli", "-p", f"{self.port}"]
+        if sentinel_cmd: run_args = run_args + ["sentinel"]
+        run_args = run_args + args
+        cp = subprocess.run(run_args, capture_output=capture_output, text=True)
+        if assert_ok:
+            assert cp.returncode == 0, f"Command failed: {run_args}"
+        return cp
```
Comment (dranikpg) [inline tests/dragonfly/sentinel_test.py:81]: I'm sure the default client libraries support sentinel commands which would have also saved you from writing `stdout_as_list_of_dicts`

**Review by dranikpg — APPROVED** [review]

File: tests/dragonfly/sentinel_test.py (hunk 1 above)
Comment (ashotland) [reply tests/dragonfly/sentinel_test.py:160]: OK, returning a coroutine makes sense to me, I'll merge this version and try that later.

File: tests/dragonfly/sentinel_test.py (hunk 2 above)
Comment (ashotland) [reply tests/dragonfly/sentinel_test.py:81]: From a brief look at the client library API it does't seem to cover all commands I've used here and I think we'd want to be able to query the internal state of the sentinel to better asses the integration.

**Pull request merged**